### PR TITLE
Integration specs for views and fix n+1 💯 

### PR DESCRIPTION
**Pull request by leehaney254**
- Solved the N+1 problem when fetching all posts
  - before
  - see-all-posts
  - After
  - see-all-eager
- Solved the N+1 problem when fetching all posts
  - before
  - see-all-comments
  - after
  - comments-eager
- Created tests for the flow and things of:
  - User index page
  - User show page
  - User post index page
  - User post show page
  - Create form
  - Comment
  - Like

### CODE REVIEW

File: spec/views/users/index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    expect(page).to have_content(@user1.posts_counter)
+    expect(page).to have_content(@user2.posts_counter)
+    expect(page).to have_content(@user3.posts_counter)
+
+    expect(page).to have_link('user1', href: user_path(@user1))
+    expect(page).to have_link('user2', href: user_path(@user2))
+    expect(page).to have_link('user3', href: user_path(@user3))
+  end
+end
```
Comment (TedLivist) [inline spec/views/users/index_spec.rb]: - [ ] Kindly look at your tests in the `spec/views/posts/index_spec.rb` file, You can see how the scenarios are divided with messages describing what the expectation of the tests are. Kindly do the same here as well.

  Ensure that all the tests outlined in the requirements for the user index page are present

File: spec/views/users/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    expect(page).to have_content('user1')
+    expect(page).to have_selector('.user-photo', count: 1)
+    expect(page).to have_content(@user1.posts_counter)
+    expect(page).to have_content(@user1.bio)
+    expect(page).to have_link('See all Posts', href: user_posts_path(@user1.id))
+    expect(page).to have_content('Post2')
+    expect(page).to have_content('Post3')
+    expect(page).to have_content('Post4')
+    expect(page).to_not have_content('Post1')
```
Comment (TedLivist) [inline spec/views/users/show_spec.rb]: - [x] Kindly look at your tests in the `spec/views/posts/index_spec.rb` file, You can see how the scenarios are divided with messages describing what the expectation of the tests are. Kindly do the same here as well.

  Ensure that all the tests outlined in the requirements for the user show page are present

File: spec/views/posts/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  scenario 'index page has no of comments and likes on a post' do
+    assert page.has_content?('Comments: 1')
+    assert page.has_content?('Likes: 0')
+  end
+
+  scenario 'index page has pagination' do
+    assert page.has_css?('.pagy-nav')
+    assert page.has_css?('.pagination')
+  end
```
Comment (TedLivist) [inline spec/views/posts/index_spec.rb]: - [x] You are missing the test that asserts when the post is clicked it should redirect to the post show page.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @yuvenalmash and @leehaney254,

Good job so far!
✔️ Passing tests
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/posts/index_spec.rb (hunk 3 above)
Comment (yuvenalmash) [reply spec/views/posts/index_spec.rb]: Test added

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @leehaney254. and @yuvenalmash👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Well done with integration specs. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by tresorsawasawa** [comment]
Hi @yuvenalmash 

Kindly ensure that you submitted your own pull request on your end

Thanks

**Pull request merged**